jwst/cube_build/src/cube_dq_utils.c

```diff
@@ -308,76 +309,106 @@ int overlap_fov_with_spaxels(int overlap_partial,  int overlap_full,
 //________________________________________________________________________________
 // Routine to setting NIRSpec dq plane for each wavelength plane
 
-int slice_wave_plane_nirspec(int w, int slicevalue,
-		      double roiw_ave,
-		      double *zc,
-		      double *coord1, double *coord2, double *wave,
-		      double *sliceno,
-		      long ncube, int npt,
-		      double *c1_min, double *c2_min, double *c1_max, double *c2_max) {
+long match_wave_plane_nirspec(double wave_plane,
+			      double roiw_ave,
+			      double coord1[],
+			      double coord2[],
+			      double wave[],
+			      double sliceno[], 
+			      long  npt,
+			      double *c1_min, double* c2_min,
+			      double *c1_max, double *c2_max,
+			      int *match_slice){
+
   /* 
      NIRSpec dq plane is set by mapping each slice to IFU wavelength plane 
      This routine maps each slice to sky and finds the min and max coordinates on the sky
      of the slice. 
 
-     w : wavelength plane
+     wave_plane : wavelength of current  plane
      slicevalue : slice # 1 to 30 
      roiw_ave : average roiw for all wavelengths
-     zc : array of wavelengths
-     coord1 : point cloud xi values
-     coord2 : point cloud eta values
+     coord1, coord2: tangent project coordinate of pt cloud
      wave : point cloud wavelength values
-     sliceno : point cloud slice no - starts at 1
-     ncube : number of cube values
+     sliceno: slice value of point cloud.
      npt: number of point cloud elements
 
      return:
-     c1_min, c2_min, c1_max, c2_max
+     c1_min, c2_min, c1_max, c2_max, match_slice
    */
 
-  int ipt, slice, status;
-  double wave_distance, c1, c2;
-  double dvalue = 10000;
-  *c1_min = dvalue;
-  *c2_min = dvalue;
-  *c1_max = -dvalue;
-  *c2_max = -dvalue;
- 
+  long ipt;
+  double wave_distance;
+  double slice;
+  long ii = 0;
+  
+  // initialize the values
+  float minvalue = 10000.0;
+  float maxvalue = -10000.0;
+  for (int i = 0; i< 30; i++){
+    c1_min[i] = minvalue;
+    c2_min[i] = minvalue;
+
+    c1_max[i] = maxvalue;
+    c2_max[i] = maxvalue;
+
+    match_slice[i] = 0;
+  }
+
   for (ipt =0; ipt< npt ; ipt++){
-    slice = (int)sliceno[ipt];
-    wave_distance = fabs(zc[w] - wave[ipt]);
+    slice = sliceno[ipt];
 
-    // Find all the coordinates on wave slice with slice = start region
+    wave_distance = fabs(wave_plane - wave[ipt]);
+    double c1 = coord1[ipt];
+    double c2 = coord2[ipt];
 
-    if (wave_distance < roiw_ave && slice == slicevalue){
-      c1 = coord1[ipt];
-      c2 = coord2[ipt];
-      // find min, max of xi eta
-      if (c1 < *c1_min) {*c1_min = c1;}
-      if (c1 > *c1_max) {*c1_max = c1;}
-      if (c2 < *c2_min) {*c2_min = c2;}
-      if (c2 > *c2_max) {*c2_max = c2;}
-    }
-  } // end looping over point cloud
+    // Find all the coordinates that fall on wavelength plane
+    if(wave_distance < roiw_ave){
```

Comment:
Not sure what flake8 does with C code, but it might complain about lack of whitespace after the "if" (just like it's now doing all over all of our python code).